OpenRA.Mods.Common/WorldExtensions.cs

```diff
@@ -49,9 +49,11 @@ public static IEnumerable<Actor> FindActorsOnLine(this World world, WPos lineSta
 			foreach (var currActor in actorsInSquare)
 			{
 				var actorWidth = 0;
-				var shapes = currActor.TraitsImplementing<HitShape>().Where(Exts.IsTraitEnabled);
-				if (shapes.Any())
-					actorWidth = shapes.Max(h => h.Info.Type.OuterRadius.Length);
+
+				// PERF: Avoid using TraitsImplementing<HitShape> that needs to find the actor in the trait dictionary.
+				foreach (var targetPos in currActor.EnabledTargetablePositions)
+					if (targetPos is HitShape h)
```

Comment:
```suggestion
					if (targetPos is HitShape hitshape)
```
same here, this isn't a LINQ lambda anymore.